### PR TITLE
Securizing bucket tags

### PR DESCRIPTION
**Pull request by adfost**
Bucket tags and/or edit/delete bucket tags disappear when user does not have those permissions.

### CODE REVIEW

File: portal-ui/src/screens/Console/Buckets/BucketDetails/BucketSummaryPanel.tsx [hunk 1]
```diff
@@ -617,43 +617,64 @@ const BucketSummary = ({
...
+                                />
+                              </SecureComponent>
+                            );
+                          }
+                          return null;
+                        })}
+                      <SecureComponent
+                        scopes={[IAM_SCOPES.S3_PUT_BUCKET_TAGGING]}
+                        resource={bucketName}
+                        errorProps={{ disabled: true }}
```
Comment (Alevsk) [inline portal-ui/src/screens/Console/Buckets/BucketDetails/BucketSummaryPanel.tsx:660]:
```suggestion
```

Tested, works fine, just delete `errorProps` from here so this button completely disappears.

**Review by Alevsk — APPROVED** [review]
Tested, works fine

File: portal-ui/src/screens/Console/Buckets/BucketDetails/BucketSummaryPanel.tsx [hunk 2]
```diff
@@ -617,43 +617,63 @@ const BucketSummary = ({
...
+                                scopes={[IAM_SCOPES.S3_PUT_BUCKET_TAGGING]}
+                                resource={bucketName}
+                                matchAll
+                                errorProps={{
+                                  deleteIcon: null,
+                                  onDelete: null,
+                                }}
+                              >
+                                <Chip
+                                  key={`chip-${index}`}
```
Comment (bexsoft) [inline portal-ui/src/screens/Console/Buckets/BucketDetails/BucketSummaryPanel.tsx:642]: Please move this key to Secure Component
```suggestion
```

File: portal-ui/src/screens/Console/Buckets/BucketDetails/BucketSummaryPanel.tsx [hunk 3]
```diff
@@ -617,43 +617,63 @@ const BucketSummary = ({
...
+                  <tr>
+                    <td className={classes.titleCol}>Tags:</td>
+                    <td>
+                      {tagKeys &&
+                        tagKeys.map((tagKey: any, index: any) => {
+                          const tag = get(tags, `${tagKey}`, "");
+                          if (tag !== "") {
+                            return (
+                              <SecureComponent
+                                scopes={[IAM_SCOPES.S3_PUT_BUCKET_TAGGING]}
```
Comment (bexsoft) [inline portal-ui/src/screens/Console/Buckets/BucketDetails/BucketSummaryPanel.tsx:633]:
```suggestion
                                key={`chip-${index}`}
                                scopes={[IAM_SCOPES.S3_PUT_BUCKET_TAGGING]}
```

**Review by bexsoft — CHANGES_REQUESTED** [review]
Would you please help us to remove this warning?
Screen Shot 2021-11-22 at 20 45 07

Beside of that comment, LGTM

File: portal-ui/src/screens/Console/Buckets/BucketDetails/BucketSummaryPanel.tsx (hunk 3 above)
Comment (bexsoft) [reply portal-ui/src/screens/Console/Buckets/BucketDetails/BucketSummaryPanel.tsx:633]: Screen Shot 2021-11-22 at 20 46 10

**Review by prakashsvmx — APPROVED** [review]
Verified with policies for a user:
- Deny Tagging  ( No Read/Write)
- Read only Tagging ( Read)
- Full Tagging ( Read/Write)

**Review by bexsoft — APPROVED** [review]
LGTM

**Pull request merged**